plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -138,9 +140,22 @@ protected void copyTableSchema(Connection connection, String catalogName, String
     protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
     {
         ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
-        ClickHouseTableProperties.getEngine(tableMetadata.getProperties())
-                .ifPresent(value -> tableOptions.add("ENGINE =" + quoted(value)));
-        return format("CREATE TABLE %s (%s) %s", quoted(remoteTableName), join(", ", columns), join(", ", tableOptions.build()));
+        Map<String, Object> tableProperties = tableMetadata.getProperties();
+        ClickHouseEngineType engine = ClickHouseTableProperties.getEngine(tableProperties);
+        tableOptions.add("ENGINE = " + engine.getEngineType());
+        if (engine == ClickHouseEngineType.MERGETREE) {
+            // order_by property is required
+            if (ClickHouseTableProperties.getOrderBy(tableProperties).isEmpty()) {
+                throw new TrinoException(INVALID_TABLE_PROPERTY, format("%s is required for %s", ClickHouseTableProperties.ORDER_BY_PROPERTY, engine.getEngineType()));
+            }
+            else {
+                tableOptions.add("ORDER BY " + ClickHouseTableProperties.getOrderBy(tableProperties).get());
+            }
+            ClickHouseTableProperties.getPrimaryKey(tableProperties).ifPresent(value -> tableOptions.add("PRIMARY KEY " + value));
+            ClickHouseTableProperties.getPartitionBy(tableProperties).ifPresent(value -> tableOptions.add("PARTITION BY " + value));
+            ClickHouseTableProperties.getSampleBy(tableProperties).ifPresent(value -> tableOptions.add("SAMPLE BY " + value));
+        }
```

Comment:
relating to https://github.com/trinodb/trino/pull/7135#discussion_r588131187,

what happens if i issue 
```
CREATE TABLE clickhouse.default.t(a int) 
WITH (partition_by = ARRAY['x'])
```
?

I would expect some kind of failure (either from the connector, or from ClickHouse), because there is no column `x`.

Then, what happens if i issue 
```
CREATE TABLE clickhouse.default.t(a int, x int) 
WITH (engine = 'LOG', partition_by = ARRAY['x'])
```
?

I would expect some kind of failure (either from the connector, or from ClickHouse), because i asked for the table to be partitioned, but it cannot be.

Reply:
```sql
CREATE TABLE clickhouse.default.t(a int) 
WITH (partition_by = ARRAY['x'])
```

```sql
CREATE TABLE clickhouse.default.t(a int, x int) 
WITH (engine = 'LOG', partition_by = ARRAY['x'])
```

Yes, the above sqls  will both create successfully, because the default table engine `Log` will ignore all other properties.

we can pass all table property regardless table engine, it will be kind of failure from ClickHouse side。
So we take this approach？

Reply:
execute the following sql in ClickHouse directly 

```sql
create table default.t (a int)  engine = Log order by a;
```

It print the following errors:

```
ClickHouse exception, code: 36, host: 10.60.242.112, port: 18123; Code: 36, e.displayText() = DB::Exception: 
Engine Log doesn't support PARTITION_BY, PRIMARY_KEY, ORDER_BY or SAMPLE_BY clauses. 
Currently only the following engines have support for the feature: [MergeTree, ReplicatedVersionedCollapsingMergeTree, 
ReplacingMergeTree, ReplicatedSummingMergeTree, ReplicatedAggregatingMergeTree, ReplicatedCollapsingMergeTree, ReplicatedGraphiteMergeTree, ReplicatedMergeTree, 
ReplicatedReplacingMergeTree, VersionedCollapsingMergeTree, SummingMergeTree,
GraphiteMergeTree, CollapsingMergeTree, AggregatingMergeTree] (version 20.3.5.21 (official build))
```

Reply:
modified codes:

```java
    @Override
    protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
    {
        ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
        Map<String, Object> tableProperties = tableMetadata.getProperties();
        ClickHouseEngineType engine = ClickHouseTableProperties.getEngine(tableProperties);
        tableOptions.add("ENGINE = " + engine.getEngineType());
        if (engine == ClickHouseEngineType.MERGETREE && formatProperty(ClickHouseTableProperties.getOrderBy(tableProperties)).isEmpty()) {
            // order_by property is required
            throw new TrinoException(INVALID_TABLE_PROPERTY, format("The property of %s is required for table engine %s", ClickHouseTableProperties.ORDER_BY_PROPERTY, engine.getEngineType()));
        }
        formatProperty(ClickHouseTableProperties.getOrderBy(tableProperties)).ifPresent(value -> tableOptions.add("ORDER BY " + value));
        formatProperty(ClickHouseTableProperties.getPrimaryKey(tableProperties)).ifPresent(value -> tableOptions.add("PRIMARY KEY " + value));
        formatProperty(ClickHouseTableProperties.getPartitionBy(tableProperties)).ifPresent(value -> tableOptions.add("PARTITION BY " + value));
        ClickHouseTableProperties.getSampleBy(tableProperties).ifPresent(value -> tableOptions.add("SAMPLE BY " + value));

        return format("CREATE TABLE %s (%s) %s", quoted(remoteTableName), join(", ", columns), join(" ", tableOptions.build()));
    }
```